tests/lava/proc/io/test_extractor.py

```diff
@@ -392,7 +399,11 @@ def _test_receive_not_empty_policy(receive_not_empty: utils.ReceiveNotEmpty,
         run_cfg = Loihi2SimCfg()
 
         extractor.run(condition=run_condition, run_cfg=run_cfg)
+        if verbose:
+            print(f'{extractor.runtime._is_running=}')
         recv_data = [extractor.receive(), extractor.receive()]
+        if verbose:
+            print(f'{recv_data=}')
         extractor.stop()
 
         return np.array(recv_data)
```

Comment:
`test_receive_data_receive_not_empty_fifo` and `test_receive_data_receive_not_empty_accumulate` appear to fail because `self.runtime._is_running` is not true at time of `extractor.receive()` and thus the check @tim-shea put in `can_receive()` fails, thus the correct `qsize` is not used when calling internal `_recieve_when*`.

``` 
def can_receive(self) -> int:
        if self.runtime is not None and \
            self.runtime._is_running:
            return self._pm_to_p_dst_port._queue.qsize()
        else:
            return 0
```